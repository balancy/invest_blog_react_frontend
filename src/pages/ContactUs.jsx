import { sendForm } from 'emailjs-com';
import React from 'react';
import { Form, Input, TextArea } from 'semantic-ui-react';
import swal from 'sweetalert';
import MainButton from '../components/UI/buttons/MainButton';

const SERVICE_ID = process.env.REACT_APP_SERVICE_ID
const TEMPLATE_ID = process.env.REACT_APP_TEMPLATE_ID
const USER_ID = process.env.REACT_APP_USER_ID

const ContactUs = () => {

    const handleOnSubmit = (e) => {
        e.preventDefault();
        sendForm(SERVICE_ID, TEMPLATE_ID, e.target, USER_ID)
            .then((result) => {
                console.log(result.text);
                swal('Успех', 'Письмо отправлено', 'success')
            }, (error) => {
                console.log(error.text);
                swal('Ошибка', 'Что-то пошло не так', 'error')
            });
        e.target.reset()
    };

    return (
        <div>
            <h1>Свяжитесь с нами</h1>

            <div className="contactPage">
                <Form className="contactForm" onSubmit={handleOnSubmit}>
                    <Form.Field
                        className='contactForm__control'
                        control={Input}
                        name='user_email'
                        placeholder='Ваш email'
                        required
                    />
                    <Form.Field
                        className='contactForm__control'
                        control={Input}
                        name='user_name'
                        placeholder='Ваше имя'
                        required
                    />
                    <Form.Field
                        className='contactForm__control'
                        control={TextArea}
                        name='user_message'
                        placeholder='Сообщение'
                        required
                    />
                    <MainButton type='submit'>Отправить</MainButton>
                </Form>
            </div >
        </div>
    )
}

export default ContactUs
