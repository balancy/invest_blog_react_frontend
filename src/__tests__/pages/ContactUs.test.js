import '@testing-library/jest-dom/extend-expect';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { configure, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import ContactUs from "../../pages/ContactUs";


configure({ adapter: new Adapter() });

test('should render ContactUs page', () => {
    const wrapper = mount(
        <ContactUs />
    );

    const h1 = wrapper.find('h1')
    const button = wrapper.find('MainButton')

    expect(h1.text()).toBe('Свяжитесь с нами')
    expect(button.text()).toBe('Отправить')
})

test('matches snapshot', () => {
    const tree = renderer
        .create(
            <ContactUs />
        ).toJSON()
    expect(tree).toMatchSnapshot()
})