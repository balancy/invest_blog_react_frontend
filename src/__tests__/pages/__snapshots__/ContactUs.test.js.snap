// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <h1>
    Свяжитесь с нами
  </h1>
  <div
    className="contactPage"
  >
    <form
      className="ui form contactForm"
      onSubmit={[Function]}
    >
      <div
        className="required field contactForm__control"
      >
        <div
          className="ui input"
        >
          <input
            aria-describedby={null}
            name="user_email"
            onChange={[Function]}
            placeholder="Ваш email"
            required={true}
            type="text"
          />
        </div>
      </div>
      <div
        className="required field contactForm__control"
      >
        <div
          className="ui input"
        >
          <input
            aria-describedby={null}
            name="user_name"
            onChange={[Function]}
            placeholder="Ваше имя"
            required={true}
            type="text"
          />
        </div>
      </div>
      <div
        className="required field contactForm__control"
      >
        <textarea
          aria-describedby={null}
          name="user_message"
          onChange={[Function]}
          onInput={[Function]}
          placeholder="Сообщение"
          required={true}
          rows={3}
        />
      </div>
      <button
        className="MainButton"
        type="submit"
      >
        Отправить
      </button>
    </form>
  </div>
</div>
`;
